### DISCUSSION TITLE
Question About Sync

### DISCUSSION THREAD

**Discussion by ClwnMan76** [ankit/stylebot#727] [Q&A]
Forgive my ignorance. If I sync from a different laptop is it going to download the latest from the cloud or upload from the laptop that doesn't have the newest changes?

**Comment by joshseltzer** [comment]
@ClwnMan76 if you have changes on the cloud, as well as local changes on your laptop that haven't been synced, then to my understanding Stylebot will attempt to merge them and then update both. This is the condition that gets [triggered here](https://github.com/ankit/stylebot/blob/main/src/sync/google-drive/sync.ts#L125). Due to the simple implementation of [merge-styles](https://github.com/ankit/stylebot/blob/main/src/sync/google-drive/merge-styles.ts#L12) though, I believe practically speaking this means that, for any particular site if you have local modifications as well as modifications from remote sync, whichever is more recent will be preserved. (Looking closely at the code, I am wondering if there is maybe a [bug in this line](https://github.com/ankit/stylebot/blob/main/src/sync/google-drive/merge-styles.ts#L28) where it will always choose the local modification if available?)
